src/cpp/cpp_typecheck_constructor.cpp

```diff
@@ -69,9 +69,8 @@ static void copy_member(
   op1.add_source_location()=source_location;
 
   side_effect_exprt assign(ID_assign, typet(), source_location);
-  assign.copy_to_operands(op0.as_expr());
-  assign.op0().add_source_location() = source_location;
-  assign.copy_to_operands(op1);
+  assign.copy_to_operands(op0.as_expr(), op1);
```

Comment:
Could use the `side_effect_exprt` constructor that also takes operands?

Reply:
indeed, done